libcaf_net/caf/net/transport_base.hpp

```diff
@@ -191,26 +189,26 @@ class transport_base {
   /// transport.
   /// @param id The id of the destination endpoint.
   /// @param buffers Pointers to the buffers that make up the packet content.
-  virtual void write_packet(id_type id, span<buffer_type*> buffers) = 0;
+  virtual void write_packet(id_type id, span<byte_buffer*> buffers) = 0;
 
   // -- buffer management ------------------------------------------------------
 
   /// Returns the next cached header buffer or creates a new one if no buffers
   /// are cached.
-  buffer_type next_header_buffer() {
+  byte_buffer next_header_buffer() {
     return next_buffer_impl(header_bufs_);
```

Comment:
This line calls the copy or move constructor and the compiler needs to know the memory layout for returning an object of type `byte_buffer` in the first place. Hence, we also need to include `byte_buffer.hpp` in this file. That alone would require the full type definition, but of course we also have a member of type `byte_buffer`.